docs/tutorials/custom_experiment.rst

```diff
@@ -0,0 +1,362 @@
+
+The analysis uses the applied Pauli frame of a randomized
+measurement experiment to de-randomize the measured counts
+and combine across samples to return a single counts dictionary
+the original circuit. This has the effect of Pauli-twirling and symmetrizing the
+measurement readout error. 
+
+.. jupyter-execute::
+
+  from numpy.random import default_rng, Generator
+  from qiskit import QuantumCircuit
+  from qiskit_experiments.framework import BaseExperiment
+  from qiskit.quantum_info import random_pauli_list
+
+  class RandomizedMeasurement(BaseExperiment):
+    """Randomized measurement experiment.
+    """
+
+    def __init__(
+        self,
+        circuit,
+        measured_qubits=None,
+        physical_qubits=None,
+        backend=None,
+        **experiment_options
+    ):
+        """Basic randomize Z-basis measurement via a Pauli frame transformation
+        
+        Note this will just append a new set of measurment at the end of a circuit.
+        A more advanced version of this experiment would be use a transpiler pass to
+        replace all exisiting measurements in a circuit with randomized measurements.
+        """
+        if physical_qubits is None:
+            physical_qubits = tuple(range(circuit.num_qubits))
+        if measured_qubits is None:
+            measured_qubits = tuple(range(circuit.num_qubits))
+        
+        # Initialize BaseExperiment
+        analysis = RandomizedMeasurementAnalysis()
+        super().__init__(physical_qubits, analysis=analysis, backend=backend)
+        
+        # Add experiment properties
+        self._circuit = circuit        
+        self._measured_qubits = measured_qubits
+        
+        # Set any init optinos
+        self.set_experiment_options(**experiment_options)
+
+    @classmethod
+    def _default_experiment_options(cls):
+        options = super()._default_experiment_options()
+        options.num_samples = "default"
+        options.seed = None
+        return options
+    
+    def circuits(self):
+        # Number of classical bits for original circuit and added measurements
+        circ_nc = self._circuit.num_clbits
+        meas_nc = len(self._measured_qubits)
+        circ_qubits = list(range(self.num_qubits))
+        circ_clbits = list(range(circ_nc))
+        meas_qubits = self._measured_qubits
+        meas_clbits = list(range(circ_nc, circ_nc + meas_nc))
+
+        # Get number of samples from options
+        num_samples = self.experiment_options.num_samples
+        if num_samples == "default":
+            num_samples = 2 ** self.num_qubits
+        
+        # Get rng seed
+        seed = self.experiment_options.seed
+        if isinstance(seed, Generator):
+            rng = seed
+        else:
+            rng = default_rng(seed)
+
+        # Sample Paulis this might have duplicates, but we don't really
+        # have any easy way of running different number of shots per circuit
+        # so we just run repeat circuits multiple times
```

Comment:
This sentence is not clear and has grammatical errors

Reply:
I think this is getting too detailed anyways and isn't about how to write a custom experiment so I removed it.